opentelemetry-exporter-gcp-logging/tests/test_cloud_logging.py

```diff
@@ -0,0 +1,177 @@
+    no_default_logname.export(
+        [
+            LogData(
+                log_record=LogRecord(
+                    body="abc",
+                    resource=Resource({}),
+                    attributes={str(i): "i" * 10000 for i in range(1000)},
+                ),
+                instrumentation_scope=InstrumentationScope("test"),
+            )
+        ]
+    )
+    assert len(caplog.records) == 1
+    assert (
+        "exceeds Cloud Logging's maximum limit of 256000 bytes" in caplog.text
+    )
+
+
+def test_convert_otlp_dict_body(
+    cloudloggingfake: CloudLoggingFake,
+    snapshot_writelogentrycalls: List[WriteLogEntriesCall],
+) -> None:
+    log_data = [
+        LogData(
+            log_record=LogRecord(
+                timestamp=1736976310997977393,
+                severity_number=SeverityNumber(20),
+                trace_id=25,
+                span_id=22,
+                attributes={
+                    "gen_ai.system": "openai",
+                    "event.name": "gen_ai.system.message",
+                },
+                body={
+                    "kvlistValue": {
+                        "values": [
+                            {
+                                "key": "content",
+                                "value": {
+                                    "stringValue": "You're a helpful assistant."
+                                },
+                            }
+                        ]
+                    }
+                },
+            ),
+            instrumentation_scope=InstrumentationScope("test"),
+        )
+    ]
+    cloudloggingfake.exporter.export(log_data)
+    assert cloudloggingfake.get_calls() == snapshot_writelogentrycalls
+    for call in cloudloggingfake.get_calls():
+        assert (
+            re.match(
+                r"^opentelemetry-python \S+; google-cloud-logging-exporter \S+ grpc-python/\S+",
+                call.user_agent,
+            )
+            is not None
+        )
+
+
+def test_convert_otlp_various_different_types_in_attrs_and_bytes_body(
+    cloudloggingfake: CloudLoggingFake,
+    snapshot_writelogentrycalls: List[WriteLogEntriesCall],
+) -> None:
+    log_data = [
+        LogData(
+            log_record=LogRecord(
+                timestamp=1736976310997977393,
+                attributes={
+                    "int": 25,
+                    "float": 25.43231,
+                    "intArray": [21, 18, 23, 17],
+                    "boolArray": [True, False, True, True],
+                },
+                body=b'{"Date": "2016-05-21T21:35:40Z", "CreationDate": "2012-05-05", "LogoType": "png", "Ref": 164611595, "Classe": ["Email addresses", "Passwords"],"Link":"http://some_link.com"}',
+            ),
+            instrumentation_scope=InstrumentationScope("test"),
+        )
+    ]
+    cloudloggingfake.exporter.export(log_data)
+    assert cloudloggingfake.get_calls() == snapshot_writelogentrycalls
+
+
+def test_convert_non_json_dict_bytes(
```

Comment:
I think there is another missing case when bytes isn't UTF8. I think it will cause decode error and nothing will get exporter.

I'm guessing the collector just sends a base64 encoded string in body in this case but I'm not 100% sure